numpy/distutils/fcompiler/environment.py

```diff
@@ -57,6 +58,12 @@ def _get_var(self, name, conf_desc):
                 if var and append and os.environ.get('NPY_DISTUTILS_APPEND_FLAGS', '0') == '1':
                     var = var + [envvar_contents]
                 else:
+                    if envvar in ['LDFLAGS', 'LDSHARED']:
+                        msg = "{} is used as is, not appended ".format(envvar) + \
+                              "to flags already defined " + \
+                              "by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 " + \
+                              "to obtain appending behavior instead"
```

Comment:
Maybe warn about `NPY_DISTUTILS_APPEND_FLAGS=1` being default in a future release?

Reply:
good point, done